Sources/FluentKit/Properties/ID.swift

```diff
@@ -71,8 +71,27 @@ public final class ID<Value>: AnyID, FieldRepresentable
         case .database:
             self.inputValue = .default
         case .random:
-            let generatable = Value.self as! (RandomGeneratable & Encodable).Type
-            self.inputValue = .bind(generatable.generateRandom())
+            // only generate an id if none is set
+            let generate: Bool
+
+            // check to see if an id has been set
+            switch inputValue {
+            case .some(let value):
+                switch value {
+                case .bind(let value):
+                    generate = (value as? Value) == nil
+                default:
+                    generate = true
+                }
+            case .none:
+                generate = true
+            }
```

Comment:
I think this snippet could be made a little more concise:

```swift
if case let .some(.bind(value)) = self.inputValue {
    generate = (value as? Value) == nil
} else {
    generate = true
}
```

You might be able to remove the `.some` from the `if` statement. Not sure.

Reply:
I was considering that, but I'm imagining more cases might need to be added in the future, so having a `switch` already there could be nice.